votingsystem/src/voting.rs

```diff
@@ -0,0 +1,419 @@
+}
+
+impl Context {
+    fn storage(&self) -> &dyn SubStorageAccess {
+        self.storage.as_ref().unwrap().as_ref()
+    }
+
+    fn storage_mut(&mut self) -> &mut dyn SubStorageAccess {
+        self.storage.as_mut().unwrap().as_mut()
+    }
+
+    fn meeting(&self) -> &dyn GeneralMeetingManager {
+        self.general_meeting.as_ref().unwrap().as_ref()
+    }
+
+    fn meeting_mut(&mut self) -> &mut dyn GeneralMeetingManager {
+        self.general_meeting.as_mut().unwrap().as_mut()
+    }
+}
+
+impl Service for Context {}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Error {
+    InvalidVote,
+}
+
+#[service]
+pub trait VoteManager: Service {
+    fn get_vote(&self, vote_id: &VoteId) -> Result<Vote, Error>;
+    fn get_shares(&self, vote_id: &VoteId) -> Result<u32, Error>;
+}
+
+impl VoteManager for Context {
+    fn get_vote(&self, vote_id: &VoteId) -> Result<Vote, Error> {
+        if !self.storage().has(vote_id.as_ref()) {
+            return Err(Error::InvalidVote)
+        }
+        let vote: Vote = serde_cbor::from_slice(&self.storage().get(vote_id.as_ref()).unwrap()).unwrap();
+        Ok((vote))
+    }
+
+    fn get_shares(&self, vote_id: &VoteId) -> Result<u32, Error> {
+        let vote: Vote = serde_cbor::from_slice(&self.storage().get(vote_id.as_ref()).unwrap()).unwrap();
+        let vote_paper_id = vote.corresponding_paper_id;
+        let vote_paper: VotePaper =
+            serde_cbor::from_slice(&self.storage().get(vote_paper_id.0.as_ref()).unwrap()).unwrap();
+        Ok(vote_paper.number_of_shares)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxCreateVotePaper {
+    general_meeting_id: GeneralMeetingId,
+    agenda_number: u32,
+    number_of_shares: u32,
+    voter_name: String,
+    voter_public_key: Public,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxVote {
+    vote_paper_id: H256,
+    choice: String,
+    voter_signature: Signature,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum TxAction {
+    CreateVotePaper(TxCreateVotePaper),
+    Vote(TxVote),
+}
+
+impl Action for TxAction {}
+pub type OwnTransaction = crate::common::SignedTransaction<TxAction>;
+
+impl Context {
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<H256, ExecuteError> {
+        if transaction.tx_type() != "Vote" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction =
+            serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign)?;
+
+        match tx.tx.action {
+            TxAction::CreateVotePaper(TxCreateVotePaper {
+                general_meeting_id,
+                agenda_number,
+                number_of_shares,
+                voter_name: voter_name,
+                voter_public_key,
+            }) => {
+                let meeting: GeneralMeeting = self.meeting_mut().get_meeting(&general_meeting_id).unwrap();
```

Comment:
Please do not use `unwrap` here.
It will make Voting module crash when the CreateVotePaper transaction has an invalid general_meeting_id.